compiler/rustc_trait_selection/src/traits/project.rs

```diff
@@ -850,65 +853,76 @@ fn opt_normalize_projection_type<'a, 'b, 'tcx>(
     // bounds. It might be the case that we want two distinct caches,
     // or else another kind of cache entry.
 
-    let cache_result = if use_cache {
-        infcx.inner.borrow_mut().projection_cache().try_start(cache_key)
-    } else {
-        Ok(())
-    };
-    match cache_result {
-        Ok(()) => debug!("no cache"),
-        Err(ProjectionCacheEntry::Ambiguous) => {
-            // If we found ambiguity the last time, that means we will continue
-            // to do so until some type in the key changes (and we know it
-            // hasn't, because we just fully resolved it).
-            debug!("found cache entry: ambiguous");
-            return Ok(None);
-        }
-        Err(ProjectionCacheEntry::InProgress) => {
-            // Under lazy normalization, this can arise when
-            // bootstrapping.  That is, imagine an environment with a
-            // where-clause like `A::B == u32`. Now, if we are asked
-            // to normalize `A::B`, we will want to check the
-            // where-clauses in scope. So we will try to unify `A::B`
-            // with `A::B`, which can trigger a recursive
-            // normalization.
-
-            debug!("found cache entry: in-progress");
-
-            // Cache that normalizing this projection resulted in a cycle. This
-            // should ensure that, unless this happens within a snapshot that's
-            // rolled back, fulfillment or evaluation will notice the cycle.
+    if use_cache {
```

Comment:
I think you can change this to `if !use_cache { return Ok(()) }` and not have to add the additional indentation.

Reply:
If the cache is not used (or when there's a cache miss) then the code below my changes is executed.